boards.txt

```diff
@@ -35828,4 +35828,155 @@ Geekble_ESP32C3.menu.EraseFlash.none.upload.erase_cmd=
 Geekble_ESP32C3.menu.EraseFlash.all=Enabled
 Geekble_ESP32C3.menu.EraseFlash.all.upload.erase_cmd=-e
 
-##############################################################
\ No newline at end of file
+##############################################################
+
+waveshare_esp32s3_touch_lcd_128.name=Waveshare ESP32S3 Touch LCD 128
+waveshare_esp32s3_touch_lcd_128.vid.0=0x1a86
+waveshare_esp32s3_touch_lcd_128.pid.0=0x55d3
+
+waveshare_esp32s3_touch_lcd_128.upload.tool=esptool_py
+waveshare_esp32s3_touch_lcd_128.upload.tool.default=esptool_py
+waveshare_esp32s3_touch_lcd_128.upload.tool.network=esp_ota
+waveshare_esp32s3_touch_lcd_128.upload.maximum_size=16777216
+waveshare_esp32s3_touch_lcd_128.upload.maximum_data_size=327680
+waveshare_esp32s3_touch_lcd_128.upload.wait_for_upload_port=true
+waveshare_esp32s3_touch_lcd_128.upload.speed=460800
+waveshare_esp32s3_touch_lcd_128.upload.flags=
+waveshare_esp32s3_touch_lcd_128.upload.extra_flags=
+
+waveshare_esp32s3_touch_lcd_128.bootloader.tool=esptool_py
+waveshare_esp32s3_touch_lcd_128.bootloader.tool.default=esptool_py
+
+waveshare_esp32s3_touch_lcd_128.serial.disableDTR=true
+waveshare_esp32s3_touch_lcd_128.serial.disableRTS=true
+
+waveshare_esp32s3_touch_lcd_128.build.tarch=xtensa
+waveshare_esp32s3_touch_lcd_128.build.bootloader_addr=0x0
+waveshare_esp32s3_touch_lcd_128.build.mcu=esp32s3
+waveshare_esp32s3_touch_lcd_128.build.core=esp32
+waveshare_esp32s3_touch_lcd_128.build.target=esp32s3
+waveshare_esp32s3_touch_lcd_128.build.variant=waveshare_esp32s3_touch_lcd_128
+waveshare_esp32s3_touch_lcd_128.build.board=WAVESHARE_ESP32S3_TOUCH_LCD_128
+
+waveshare_esp32s3_touch_lcd_128.build.f_cpu=240000000L
+waveshare_esp32s3_touch_lcd_128.build.flash_size=16MB
+waveshare_esp32s3_touch_lcd_128.build.flash_freq=80m
+waveshare_esp32s3_touch_lcd_128.build.flash_mode=dio
+waveshare_esp32s3_touch_lcd_128.build.boot=dio
+waveshare_esp32s3_touch_lcd_128.build.partitions=default
+
```

Comment:
```suggestion

waveshare_esp32s3_touch_lcd_128.menu.USBMode.hwcdc=Hardware CDC and JTAG
waveshare_esp32s3_touch_lcd_128.menu.USBMode.hwcdc.build.usb_mode=1
waveshare_esp32s3_touch_lcd_128.menu.USBMode.hwcdc.upload.use_1200bps_touch=false
waveshare_esp32s3_touch_lcd_128.menu.USBMode.hwcdc.upload.wait_for_upload_port=false
waveshare_esp32s3_touch_lcd_128.menu.USBMode.default=USB-OTG
waveshare_esp32s3_touch_lcd_128.menu.USBMode.default.build.usb_mode=0
waveshare_esp32s3_touch_lcd_128.menu.USBMode.default.upload.use_1200bps_touch=true
waveshare_esp32s3_touch_lcd_128.menu.USBMode.default.upload.wait_for_upload_port=true

waveshare_esp32s3_touch_lcd_128.menu.MSCOnBoot.default=Disabled
waveshare_esp32s3_touch_lcd_128.menu.MSCOnBoot.default.build.msc_on_boot=0
waveshare_esp32s3_touch_lcd_128.menu.MSCOnBoot.msc=Enabled (Requires USB-OTG Mode)
waveshare_esp32s3_touch_lcd_128.menu.MSCOnBoot.msc.build.msc_on_boot=1

waveshare_esp32s3_touch_lcd_128.menu.DFUOnBoot.default=Disabled
waveshare_esp32s3_touch_lcd_128.menu.DFUOnBoot.default.build.dfu_on_boot=0
waveshare_esp32s3_touch_lcd_128.menu.DFUOnBoot.dfu=Enabled (Requires USB-OTG Mode)
waveshare_esp32s3_touch_lcd_128.menu.DFUOnBoot.dfu.build.dfu_on_boot=1

```

Reply:
Made all changes based on suggestions. Will test later this evening.

Reply:
Tested changes and it's compiling/running well.